test/rand_test.c

```diff
@@ -44,10 +45,42 @@ static int test_rand(void)
     return 1;
 }
 
+static int test_rand_uniform(void)
+{
+    uint32_t x, i, j;
+    int err = 0, res = 0;
+    OSSL_LIB_CTX *ctx;
+
+    if (!test_get_libctx(&ctx, NULL, NULL, NULL, NULL))
+        goto err;
+
+    for (i = 1; i < 100; i += 13) {
+        x = ossl_rand_uniform_uint32(ctx, i, &err);
+        if (!TEST_int_eq(err, 0)
+                || !TEST_uint_ge(x, 0)
+                || !TEST_uint_lt(x, i))
+            return 0;
+    }
+    for (i = 1; i < 100; i += 17)
+        for (j = i + 1; j < 150; j += 11) {
+            x = ossl_rand_range_uint32(ctx, i, j, &err);
+            if (!TEST_int_eq(err, 0)
+                    || !TEST_uint_ge(x, i)
+                    || !TEST_uint_lt(x, j))
+                return 0;
+        }
```

Comment:
Should there be negative tests? i.e. that we get an error if we call either function with invalid parameters?

Reply:
They'll abort due to the assert.  So no.

Reply:
It would be nice if there was a way in the test framework to have "expect abort", so these sort of things could be tested, but that's not for this PR